internal/serverinstall/ecs.go

```diff
@@ -0,0 +1,2190 @@
+
+			return nil
+		},
+
+		Run: func(s LifecycleStatus) error {
+			runSvcArn, err = i.LaunchRunner(
+				ctx, s, log, ui, sess,
+				opts.AdvertiseClient.Env(),
+				executionRole,
+				logGroup,
+				ulid,
+			)
+			return err
+		},
+
+		Cleanup: func(s LifecycleStatus) error { return nil },
+	}
+
+	if err := lf.Execute(log, ui); err != nil {
+		return err
+	}
+
+	s.Update("Created ECS Service (%s)", runnerName)
+	log.Debug("runner service started", "arn", *runSvcArn)
+
+	s.Done()
+
+	return nil
+}
+
+var defaultResourceQuery = "{\"ResourceTypeFilters\":[\"AWS::AllSupported\"],\"TagFilters\":[{\"Key\":\"%s\",\"Values\":[]}]}"
+
+func (i *ECSInstaller) UninstallRunner(
+	ctx context.Context,
+	opts *InstallOpts,
+) error {
+	ui := opts.UI
+	log := opts.Log
+	s := new(lStatus)
+	s.ui = ui
+	defer s.Close()
+	s.Status("Uninstalling Runner resources...")
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: i.config.region,
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+	rgSvc := resourcegroups.New(sess)
+
+	query := fmt.Sprintf(defaultResourceQuery, runnerName)
+	results, err := rgSvc.SearchResources(&resourcegroups.SearchResourcesInput{
+		ResourceQuery: &resourcegroups.ResourceQuery{
+			Type:  aws.String(resourcegroups.QueryTypeTagFilters10),
+			Query: aws.String(query),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	resources := results.ResourceIdentifiers
+	var clusterArn string
+	for _, r := range resources {
+		if *r.ResourceType == "AWS::ECS::Cluster" {
+			clusterArn = *r.ResourceArn
+		}
+	}
+	s.Status("Deleting ECS resources...")
+	if err := deleteEcsCommonResources(ctx, s, sess, clusterArn, resources); err != nil {
+		return err
+	}
+	s.Status("Deleting Cloud Watch Log Group resources...")
+	if err := deleteCWLResources(ctx, s, sess, defaultRunnerLogGroup); err != nil {
+		return err
+	}
+	return nil
+}
+
+// HasRunner implements Installer.
+func (i *ECSInstaller) HasRunner(
+	ctx context.Context,
+	opts *InstallOpts,
+) (bool, error) {
+	log := opts.Log
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: i.config.region,
+		Logger: log,
+	})
+	if err != nil {
+		return false, err
+	}
+	ecsSvc := ecs.New(sess)
+	// query what subnets and vpc information from the server service
+	services, err := ecsSvc.DescribeServices(&ecs.DescribeServicesInput{
+		Cluster:  aws.String(i.config.cluster),
+		Services: []*string{aws.String(runnerName)},
+	})
+	if err != nil {
+		return false, err
+	}
+
+	return len(services.Services) > 0, nil
+}
+
+func (i *ECSInstaller) InstallFlags(set *flag.Set) {
+	set.StringVar(&flag.StringVar{
+		Name:    "ecs-cluster",
+		Target:  &i.config.cluster,
+		Usage:   "Configures the Cluster to install into.",
+		Default: "waypoint-server",
+	})
+	set.StringVar(&flag.StringVar{
+		Name:    "region",
+		Target:  &i.config.region,
+		Usage:   "Configures the region specific things.",
+		Default: "us-west-2",
+	})
+	set.StringSliceVar(&flag.StringSliceVar{
+		Name:   "subnets",
```

Comment:
I rushed to post these during our sync but for the outside observer, we made a decision early on for serverinstall flags to always be namespaced with the platform they're for, so that's the reasoning here 😄

Reply:
I updated the names to have prefixes in 815954cb